apps/aecore/lib/aecore/structures/data_tx.ex

```diff
@@ -0,0 +1,88 @@
+defmodule  Aecore.Structures.DataTx do
+  @moduledoc """
+  Aecore structure of a transaction data.
+  """
+
+  alias Aecore.Keys.Worker, as: Keys
+  alias Aeutil.Serialization
+  alias Aeutil.Parser
+
+  @typedoc "Reason for the error"
+  @type reason :: String.t()
+
+  @typedoc "Structure of the main transaction wrapper"
+  @type t :: %__MODULE__{
+    type: atom(),
+    payload: any(),
+    from_acc: binary(),
+    fee: non_neg_integer(),
+    nonce: non_neg_integer()
+  }
+
+  @doc """
+  Definition of Aecore DataTx structure
+
+  ## Parameters
+  - type: The type of transaction that may be added to the blockchain
+  - payload: The strcuture of the specified transaction type
+  - from_acc: The public address of the account originating the transaction
+  - fee: The amount of tokens given to the miner
+  - nonce: A random integer generated on initialisation of a transaction (must be unique!)
+  """
+  defstruct [:type, :payload, :from_acc, :fee, :nonce]
+  use ExConstructor
+
+  @spec init(atom(), map(), binary(), integer(), integer()) :: {:ok, DataTx.t()}
```

Comment:
is the return type correct?

Reply:
can you be more specific of the `map()` type

Reply:
Maybe not, will look into this and make the map() type more specific

Reply:
be more specific about `atom()` as well